js/deer-record.js

```diff
@@ -69,154 +69,263 @@ export default class DeerReport {
         this.evidence = elem.getAttribute(DEER.EVIDENCE) // inherited to inputs
         this.context = elem.getAttribute(DEER.CONTEXT) // inherited to inputs
         this.type = elem.getAttribute(DEER.TYPE)
-        this.inputs = document.querySelectorAll(DEER.INPUTS.map(s=>s+"["+DEER.KEY+"]").join(","))
-
-        Array.from(this.inputs).forEach(inpt=>inpt.addEventListener('input', () => inpt.$isDirty = true))
-
+        this.inputs = elem.querySelectorAll(DEER.INPUTS.map(s => s + "[" + DEER.KEY + "]").join(","))
         changeLoader.observe(elem, {
-            attributes:true
+            attributes: true
         })
-        
         elem.oninput = event => this.$isDirty = true
         elem.onsubmit = this.processRecord.bind(this)
-        
+
         if (this.id) {
             //Do we want to expand for all types?
-            UTILS.expand({"@id":this.id})
-            //What if there are no annotations on it and the things I need to know are already in the object?
-            //Expand only returned an object like {"@id": "http://an/id"} instead of resolving it, which is what I expected. 
-            .then((function(obj){
-                Object.keys(obj).forEach((function(key){
+            UTILS.expand({ "@id": this.id })
+                .then((function(obj) {
                     try {
-                        for(let el of Array.from(this.inputs)) {
-                            if(el.getAttribute(DEER.KEY)===key){
-                                let assertedValue = UTILS.getValue(obj[key])
-                                if(Array.isArray(assertedValue)) {
-                                    for (const v of assertedValue) {
-                                        if(!el.value && (["string","number"].indexOf(typeof v)!==-1)){
-                                            el.value = v
+                        let inputElems = Array.from(this.inputs)
```

Comment:
It was intentional.  It is because form.input is necessarily the primary loop and the guaranteed worst case O(n) operation because you have to look at them all not matter what (not true for `obj.keys`).  I wanted to optimize the overall processing/runtime by optimizing O(n) * O(some consequential function because of something you found about input n).

The only need to do an `obj.key` lookup is if an `input[deer-key='avalue']` exists.  It is also much more likely that an `obj.key` lookup will be faster than shallow copying and looping over a set of input HTMLElements and checking against an attribute, especially since working with HTMLElements is notoriously costly.  I believe you side step a bunch of this cost by storing the elements in `this.inputs` ahead of time, which means you aren't repeating DOM look ups (smart), but still.  

`obj.key` lookup is (worst case) O(n) just like looping over each `obj.key` is O(n), but looping over each `obj.key` and then checking each `input` for something guarantees the worst case of O(n) * O(m) where `n` and `m` are `obj.keys.length` and `form.inputs.length`, respectively.  If instead we loop over each input and check that it needs an `obj.key` lookup, we could end up with the best case scenario of O(0) * O(m), which avoids risking the exponential potential (in the case where m >= n) when we don't need it.

Not only that `Array.from(this.inputs).forEach(inpt=>inpt.addEventListener('input', () => inpt.$isDirty = true))` was called before this logic, which mean `O(n) + (O(n) * O(m)` which I felt we could collapse if the priority loop was to loop the `<input>`s then add the event listener when we find it has a DEER.KEY during that loop.  

Arrays.from() in general is also not free, since it makes a shallow copy of an iterable, which also happened for each `obj.key`, which felt like extra work.

It also made the test and error/warning feedback make more sense to me.  Detecting and handling duplicates, for example, was difficult doing it the other way.

In summary,  I feel that doing it this way cuts back on some extraneous processing and makes detect-and-handle of peculiarities smoother (since peculiarities are going to be caused by devs doing something funky with `<inputs>`).   I am open to other opinions, as I do not dare call this fact without proper scientific proof.